src/library-authoring/component-picker/ComponentPicker.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Button, Stepper } from '@openedx/paragon';
+import { useSearchParams } from 'react-router-dom';
+
+import { LibraryProvider, useLibraryContext } from '../common/context';
+import LibraryAuthoringPage from '../LibraryAuthoringPage';
+import LibraryCollectionPage from '../collections/LibraryCollectionPage';
+import SelectLibrary from './SelectLibrary';
+import messages from './messages';
+
+interface LibraryComponentPickerProps {
+  returnToLibrarySelection: () => void;
+}
+
+const InnerComponentPicker: React.FC<LibraryComponentPickerProps> = ({ returnToLibrarySelection }) => {
+  const { collectionId } = useLibraryContext();
+
+  if (collectionId) {
+    return <LibraryCollectionPage />;
+  }
+  return <LibraryAuthoringPage returnToLibrarySelection={returnToLibrarySelection} />;
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export const ComponentPicker = () => {
+  const intl = useIntl();
+  const [searchParams] = useSearchParams();
+  let parentLocator = searchParams.get('parentLocator');
+
+  // istanbul ignore if: this should never happen
+  if (!parentLocator) {
+    throw new Error('parentLocator is required');
+  }
+
+  // URLSearchParams decodes '+' to ' ', so we need to convert it back
```

Comment:
This shouldn't be needed. The problem was that the ID wasn't properly encoded in the first place. The + should have been replaced with `%2B`

![Screenshot](https://github.com/user-attachments/assets/c62e773c-3bde-40db-a2cf-d3a11cc96d10)

Reply:
The Unit Outline doesn't use the encoded key in the URL, so I opted not to use it here, either.
![image](https://github.com/user-attachments/assets/97daa307-04eb-4d82-bbd6-3a97fea94ff6)

It will also work if we receive the URL encoded.
Let me know if this is an issue!

Reply:
Yes, but there it's in the path part of the URL. Here it's in a query string, and it's _definitely_ wrong to assume that a `+` in a query string value will be interpreted as `+`

Reply:
It's fine to leave this in; as you said it's not doing any harm. But we should be encoding the ID when we pass it into the query string.